cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -539,7 +542,7 @@ IceInternal::OutgoingConnectionFactory::createConnection(const TransceiverPtr& t
             _instance,
             transceiver,
             ci.connector,
-            ci.endpoint->compress(false),
+            ci.endpoint->compress(false)->timeout(-1),
```

Comment:
For compress, we insert it twice instead of "fixing" the endpoint. It would make sense to clean this up in a follow-up PR.